code/controllers/subsystem/ghost_spawns.dm

```diff
@@ -0,0 +1,216 @@
+	log_debug("Polling candidates [role ? "for [get_roletext(role)]" : "\"[question]\""] for [poll_time / 10] seconds")
+
+	polls_active = TRUE
+	total_polls++
+
+	var/datum/candidate_poll/P = new(role, question, poll_time)
+	currently_polling += P
+
+	if(!next_poll_to_finish || poll_time < next_poll_to_finish.time_left())
+		next_poll_to_finish = P
+
+	var/category = "[P.hash]_notify_action"
+
+	for(var/mob/dead/observer/M in (ignore_respawnability ? GLOB.player_list : GLOB.respawnable_list))
+		if(!is_eligible(M))
+			continue
+
+		M << 'sound/misc/notice2.ogg'
+		if(flash_window)
+			window_flash(M.client)
+
+		// If we somehow send two polls for the same mob type, but with a duration on the second one shorter than the time left on the first one,
+		// we need to keep the first one's timeout rather than use the shorter one
+		var/obj/screen/alert/notify_action/current_alert = LAZYACCESS(M.alerts, category)
+		var/alert_time = poll_time
+		var/alert_poll = P
+		if(current_alert && current_alert.timeout > (world.time + poll_time - world.tick_lag))
+			alert_time = current_alert.timeout - world.time + world.tick_lag
+			alert_poll = current_alert.poll
+
+		// Send them an on-screen alert
+		var/obj/screen/alert/notify_action/A = M.throw_alert(category, /obj/screen/alert/notify_action, timeout_override = alert_time, no_anim = TRUE)
+		if(!A)
+			continue
+
+		A.icon = ui_style2icon(M.client?.prefs.UI_style)
+		A.name = "Looking for candidates"
+		A.desc = "[question]\n\n(expires in [poll_time / 10] seconds)"
+		A.show_time_left = TRUE
+		A.poll = alert_poll
+
+		var/inherited_sign_up = FALSE
+		var/num_stack = 1
+		for(var/existing_poll in currently_polling)
+			var/datum/candidate_poll/P2 = existing_poll
+			if(P != P2 && P.hash == P2.hash)
+				// If there's already a poll for an identical mob type ongoing and the client is signed up for it, sign them up for this one
+				if(!inherited_sign_up && (M in P2.signed_up) && P.sign_up(M, TRUE))
+					A.display_signed_up()
+					inherited_sign_up = TRUE
+				// This number is used to display the number of polls the alert regroups
+				num_stack++
+		if(num_stack > 1)
+			A.display_stacks(num_stack)
+
+		// Image to display
+		var/image/I
+		if(source)
+			if(!ispath(source))
+				var/atom/S = source
+				var/old_layer = S.layer
+				var/old_plane = S.plane
+
+				S.layer = FLOAT_LAYER
+				S.plane = FLOAT_PLANE
+				A.overlays += S
+				S.layer = old_layer
+				S.plane = old_plane
+			else
+				I = image(source, layer = FLOAT_LAYER, dir = SOUTH)
+		else
+			// Just use a generic image
+			I = image('icons/effects/effects.dmi', icon_state = "static", layer = FLOAT_LAYER, dir = SOUTH)
+
+		if(I)
+			I.layer = FLOAT_LAYER
+			I.plane = FLOAT_PLANE
+			A.overlays += I
+
+		// Start processing it so it updates visually the timer
+		START_PROCESSING(SSprocessing, A)
+		A.process()
+
+	// Sleep until the time is up
+	UNTIL(P.finished)
```

Comment:
Have to be careful with this. Since if this proc is not called from a proc that has set `waitfor` to `TRUE/1` then it will hang the game.

Reply:
Gotcha, I'll add a warning above the define and I'll make sure any calls to the proc don't have `waitfor` set to `FALSE`/`0`